crates/wasmtime/src/store.rs

```diff
@@ -1496,8 +1496,109 @@ unsafe impl<T> wasmtime_runtime::Store for StoreInner<T> {
         (&mut inner.externref_activations_table, &inner.modules)
     }
 
-    fn limiter(&mut self) -> Option<&mut dyn wasmtime_runtime::ResourceLimiter> {
-        <Self>::limiter(self)
+    #[cfg(not(feature = "async"))]
+    fn limiter_memory_growing(
+        &mut self,
+        current: usize,
+        desired: usize,
+        maximum: Option<usize>,
+    ) -> bool {
+        if let Some(limiter) = <Self>::limiter(self) {
+            limiter.memory_growing(current, desired, maximum)
+        } else {
+            true
+        }
+    }
+
+    #[cfg(feature = "async")]
+    fn limiter_memory_growing(
+        &mut self,
+        current: usize,
+        desired: usize,
+        maximum: Option<usize>,
+    ) -> bool {
+        if self.async_support() {
+            let async_cx = self.async_cx();
+            if let Some(limiter) = <Self>::limiter(self) {
+                unsafe {
+                    async_cx
+                        .block_on(
+                            limiter
+                                .memory_growing_async(current, desired, maximum)
+                                .as_mut(),
+                        )
+                        .expect("limiter should never trap")
```

Comment:
I think we'll need to handle this because in this case a trap means that the future was dropped on the outside, so we need to propagate this trap to wasm to make wasm "abort" gracefully as well. Basically once the future is dropped we just need to clean it up quickly so the thinking is to propagate the error all the way back to the start.

Reply:
Good point, I meant to check this assumption by you. I forgot about the whole future dropped thing.